hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2616,6 +2617,19 @@ public Integer getWritesFileIdEncoding() {
     return props.getInteger(WRITES_FILEID_ENCODING, HoodieMetadataPayload.RECORD_INDEX_FIELD_FILEID_ENCODING_UUID);
   }
 
+  public boolean needResolveWriteConflict(WriteOperationType operationType) {
+    if (getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
+      // Skip to resolve conflict for non bulk_insert operation if using non-blocking concurrency control
+      return !isNonBlockingConcurrencyControl() || maybeBulkInsert(operationType);
+    } else {
+      return false;
+    }
+  }
+
+  private boolean maybeBulkInsert(WriteOperationType operationType) {
+    return operationType == WriteOperationType.UNKNOWN || operationType == WriteOperationType.BULK_INSERT;
```

Comment:
What if we just ignore the `WriteOperationType.UNKNOWN` operation? That means when NB-CC is enabled, the conflic resolve is skipped for it.

Reply:
I haven't made a decision on how to resolve conflicts for `writeOperationType.UNKNOWN` operation yet. Here, I've adopted a conservative approach, even when NB-CC is enabled,  `writeOperationType.UNKNOWN` operation cannot skip the conflict resolution step.

Reply:
You need to make it clear because in `[TransactionUtils.java](https://github.com/apache/hudi/pull/9896/files#diff-09f645e892253b62a07fdf3719bac75aca166da7c0758fad832d8cf855684eb6)`, you use `UNKNOWN` as the fallback type.